platform/linux/src/headless_backend_egl.cpp

```diff
@@ -10,9 +10,24 @@
 namespace mbgl {
 
 struct EGLImpl : public HeadlessBackend::Impl {
-    EGLImpl(EGLContext glContext_, EGLDisplay display_)
+    EGLImpl(EGLContext glContext_, EGLDisplay display_, EGLConfig config_)
             : glContext(glContext_),
-              display(display_) {
+              display(display_),
+              config(config_) {
+#if defined(__ANDROID__)
+        //Create a PBuffer surface (in conjunction with EGL_SURFACE_TYPE, EGL_PBUFFER_BIT)
+        const EGLint surfAttribs[] = {
+            EGL_WIDTH, 512,
+            EGL_HEIGHT, 512,
+            EGL_LARGEST_PBUFFER, EGL_TRUE,
+            EGL_NONE
+        };
+
+        glSurface = eglCreatePbufferSurface(display, config, surfAttribs);
+        if(glSurface == EGL_NO_SURFACE) {
```

Comment:
You can use `clang-format` with the formatting file in the root directory to format C++ code automatically.